effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -125,6 +125,20 @@ object Transformer {
         noteBlockParams(id, allParams)
         Def(Label(transform(id), allParams), transform(body), transform(rest))
 
+      case lifted.Def(id, tpe, block @ lifted.New(impl), rest) =>
+        // TODO freeParams?
+        // TODO deal with evidenve?
+        val interface = tpe.asInstanceOf[symbols.Interface]
+        val implTransformed = impl.clauses.sortBy[Int]({
+          case (operationName, _) =>
+            interface.ops.indexOf(operationName)
```

Comment:
While this works, in other places in the compiler we would iterate over `interface.ops` like
```
interface.ops map {
  case op => transform ... impl(op) ...
}
```
This has the advantage that we maintain the invariant that all operations are actually there and all others would be dropped in the worst case.

Reply:
Changed that (although `impl` is not a map, so it's `impl.find( ... ).get` now).